TESTS/mbed_hal/trng/main.cpp

```diff
@@ -0,0 +1,276 @@
+        result = nvstore.get(NVKEY, sizeof(buffer), buffer, actual);
+        TEST_ASSERT_EQUAL(NVSTORE_SUCCESS, result);
+#else
+        /*Using base64 to decode data sent from host*/
+        uint32_t lengthWritten = 0;
+	    uint32_t charsProcessed = 0;
+        result = esfs_DecodeNBase64((const char*)value, MSG_VALUE_LEN, buffer, BUFFER_LEN, &lengthWritten, &charsProcessed);
+        TEST_ASSERT_EQUAL(0, result);
+#endif
+        memcpy(input_buf, buffer, BUFFER_LEN);
+    }
+
+    /*Fill buffer with trng values*/
+    result = fill_buffer_trng(buffer, &trng_obj, sizeof(buffer));
+    TEST_ASSERT_EQUAL(0, result);
+
+    /*lzf_compress will try to compress the random data, if it succeeded it means the data is not really random*/
+    if (strcmp(key, MSG_TRNG_TEST_STEP1) == 0)
+    {
+        printf("\n******TRNG_TEST_STEP1*****\n");
+        out_comp_buf_len = BUFFER_LEN + (BUFFER_LEN / 4);
+        comp_res = lzf_compress((const void *)buffer, 
+                                (unsigned int)sizeof(buffer), 
+                                (void *)out_comp_buf, 
+                                out_comp_buf_len, 
+                                (unsigned char **)htab);
+        if (comp_res >= BUFFER_LEN)
+        {
+            printf("trng_get_bytes for buffer size %d was successful", sizeof(buffer));
+        }
+        else
+        {
+            printf("trng_get_bytes for buffer size %d was unsuccessful", sizeof(buffer));
+            TEST_ASSERT(false);
+        }
+        printf("\n******FINISHED_TRNG_TEST_STEP1*****\n\n");
+    }
+    else if (strcmp(key, MSG_TRNG_TEST_STEP2) == 0)
+    {
+        printf("\n******TRNG_TEST_STEP2*****\n");
+        result = fill_buffer_trng(temp_buff, &trng_obj, sizeof(temp_buff));
+        TEST_ASSERT_EQUAL(0, result);
+
+        out_comp_buf_len = 2 * BUFFER_LEN + (BUFFER_LEN / 2);
+        comp_res = lzf_compress((const void *)temp_buff, 
+                                (unsigned int)sizeof(temp_buff), 
+                                (void *)out_comp_buf, 
+                                out_comp_buf_len, 
+                                (unsigned char **)htab);
+        
+        if (comp_res >= BUFFER_LEN)
+        {
+            printf("trng_get_bytes for buffer size %d was successful", sizeof(temp_buff));
+        }
+        else
+        {
+            printf("trng_get_bytes for buffer size %d was unsuccessful", sizeof(temp_buff));
+            TEST_ASSERT(false);
+        }
+        printf("\n******FINISHED_TRNG_TEST_STEP2*****\n\n");
+
+        printf("******TRNG_TEST_STEP3*****\n");
+
+        memcpy(input_buf + BUFFER_LEN, buffer, BUFFER_LEN);
+        comp_res = lzf_compress((const void *)input_buf, 
+                                (unsigned int)sizeof(input_buf), 
+                                (void *)out_comp_buf, 
+                                out_comp_buf_len, 
+                                (unsigned char **)htab);
+
+        if (comp_res >= BUFFER_LEN)
+        {
```

Comment:
Can you fix formatting (follow mbed OS code style) , this one (`{`) should be attached

Reply:
Sorry for that, all formatting will be fixed